operator/pkg/resources/configmap.go

```diff
@@ -807,30 +813,59 @@ func (r *ConfigMapResource) SetLastAppliedConfigurationInCluster(
 	return nil
 }
 
+// PrepareSeedServerList - supports only > 22.3 (featuregates.EmptySeedStartCluster(r.pandaCluster.Spec.Version)
 func (r *ConfigMapResource) PrepareSeedServerList(cr *config.RedpandaNodeConfig) error {
-	c := r.pandaCluster.Spec.Configuration
-	replicas := r.pandaCluster.GetCurrentReplicas()
+	r.logger.Info("New seed list")
+	var addresses []string
+
+	for npName, npStatus := range r.pandaCluster.Status.NodePools {
```

Comment:
Why is this being pulled from Status? This screams race condition to me. Can we not iterate over the nodepools in the spec?

Reply:
status contains also deleted node pools. therefore, we want to include it as input for computing the seed server list.

it is an imperfect solution, that node pools are stored in spec, and not a separate CRD. we discussed this ~1.5 months back in slack. i believe as the operators converge, v2/v3 can introduce a nodepool CRD and do this better - then these cases will disappear, as a separate CRD can distinguish between "delete started" and "finally deleted".

i suppose, for clarity of the code, this function can be refactored to use "spec first", and add add nodePoolStatus-sourced pools on top - looking into refactoring this now.

Reply:
It looks like this special handling is not required here anymore. GetNodePools includes deleted ones, so no need to use status here. Updated the code (awaiting CI to run the entire test suite to confirm this is correct).

Reply:
> therefore, we want to include it as input for computing the seed server list.

Why do we need to include the (potentially) deleted nodepools in the seed list? Shouldn't it be sufficient to have a subset of all brokers in the seedlist?

Reply:
tbh, i am not 100% sure. if it is safe to do so, i would change it. so far, all brokers were included, so i am following this existing practice. 

i suspect there are corner cases, eg. if a change adds and deletes a nodepool at the same time, we do not point to the old brokers anymore, and new nodes cannot join.